OwnYourData/ClinicalTrials/ClinicalTrialsView.swift

```diff
@@ -6,47 +6,96 @@
 // SPDX-License-Identifier: MIT
 //
 
+import CoreLocation
 import OpenAPIClient
 import SpeziViews
 import SwiftUI
 
-
 struct ClinicalTrialsView: View {
     @State private var viewState: ViewState = .idle
     @State private var trials: [TrialDetail] = []
+    @State private var collapseStates: [Bool] = [] // Track collapsed state for each trial
+    @State private var isLoading: Bool = false // Track whether trials are being actively loaded
+    @State private var zipCode: String = "10027" // ZipCode search state variable, defaults to 10027
+    @State private var searchDistance: String = "100" // Distance search state variable, default 100
     
+    // Create a CLLocationManager instance to manage location services for trials match search
+    private let locationManager = CLLocationManager()
+    // Create a CLGeocoder instance to convert zip code to coordinates
+    private let geocoder = CLGeocoder()
```

Comment:
I would suggest using @vishnuravi's Spezi Location package. At some point we will clean this up, add more documentation, and consider adopting as an an official module but it is currently fully functional and simplifies a lot of the issues your are having at the moment of dealing with the async challenge of getting a location.

You should be able to import the package from the main branch of https://github.com/vishnuravi/SpeziLocation and the utilize the compact Spezi module https://github.com/vishnuravi/SpeziLocation/blob/main/Sources/SpeziLocation/SpeziLocation.swift as demonstrated in the UI testing app:
1. Add it to the Spezi Configuration: https://github.com/vishnuravi/SpeziLocation/blob/main/Tests/UITests/TestApp/TestAppDelegate.swift#L16
2. Use it in your view with a nice async API that will solve some of the race conditions you are having at the moment: https://github.com/vishnuravi/SpeziLocation/blob/main/Tests/UITests/TestApp/LocationView.swift